cot-cli/tests/snapshot_testing/migration/mod.rs

```diff
@@ -0,0 +1,116 @@
+use std::io::Write;
+
+use clap_verbosity_flag::{OffLevel, Verbosity};
+use cot_cli::migration_generator::MigrationGeneratorOptions;
+use cot_cli::{migration_generator, test_utils};
+
+use super::*;
+
+#[test]
+#[expect(clippy::cast_possible_truncation)]
+fn migration_list_empty() {
+    let temp_dir = tempfile::TempDir::with_prefix("cot-test-").unwrap();
+    test_utils::make_package(temp_dir.path()).unwrap();
+
+    let mut cmd = cot_cli!("migration", "list");
+    cmd.current_dir(temp_dir.path());
+
+    for (idx, mut cli) in cot_clis_with_verbosity(&cmd).into_iter().enumerate() {
+        let filter = Verbosity::<OffLevel>::new(idx as u8, 0).filter();
+
+        insta::with_settings!(
+            { description => format!("Verbosity level: {filter}") },
+            { assert_cmd_snapshot!(cli); }
+        );
+    }
+}
+
+#[test]
+#[expect(clippy::cast_possible_truncation)]
+fn migration_list_existing() {
+    let temp_dir = tempfile::TempDir::with_prefix("cot-test-").unwrap();
+    test_utils::make_package(temp_dir.path()).unwrap();
+    let mut main = std::fs::OpenOptions::new()
+        .append(true)
+        .open(temp_dir.path().join("src").join("main.rs"))
+        .unwrap();
+    write!(
+        main,
+        "#[model]\nstruct Test {{\n#[model(primary_key)]\nid: Auto<i32>\n}}"
```

Comment:
As with the other comment in this file, please make this code valid and readable.